### PR TITLE
SNV Frequency Table Update

### PR DESCRIPTION
**Pull request by runjin326**
#### What scientific question is your analysis addressing?
This update is to implement the changes on the SNV frequency tables per FNL feedback.

#### What was your approach?

Rename of columns and additions of 3 columns for gene-level table were added as following:
```
gene_level_mut_freq_tbl <- gene_level_mut_freq_tbl %>%
  rename(targetFromSourceId = Gene_Ensembl_ID,
         diseaseFromSourceMappedId = EFO) %>%
  mutate(datatypeId = "somatic_mutation",
         datasourceId = "chop_gene_level_snv")

# generate UUID for each row of the table
uuid_string_2 <- uuid(nrow(gene_level_mut_freq_tbl))
# make sure all the uuids generated are unique
stopifnot(length(unique(uuid_string_2)) == nrow(gene_level_mut_freq_tbl))

# assign UUID to each row
gene_level_mut_freq_tbl <- gene_level_mut_freq_tbl %>%
  dplyr::mutate(chop_uuid = uuid_string_2)
```

Rename of columns and additions of 3 columns for variant-level table were added similarly with the field `datasourceId` as `chop_variant_level_snv`.

#### What GitHub issue does your pull request address?
PediatricOpenTargets/ticket-tracker#197

#### Which areas should receive a particularly close look?
R package `ids` was used to generate UUID - please run locally to see whether we need to add one line to install the package (it ran for me without having to install it but just want to be sure).

#### Is there anything that you want to discuss further?
No. 

#### Is the analysis in a mature enough form that the resulting figure(s) and/or table(s) are ready for review?
No - review after v10 release.

#### What types of results are included (e.g., table, figure)?
TSV and JSONL files.

#### What is your summary of the results?
- With the current code change, the number of entries in each file did not change.
  - Original gene-level and variant-level tables have `100017` and `413349` rows respectively and these numbers remain the same with the new files generated.
- For both tables. 3 more columns datatypeId, chop_uuid and datasourceId were added and I confirmed they have expected values as indicated in the ticket.

#### Reproducibility Checklist

- [x] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [ ] This analysis has been added to continuous integration.

#### Documentation Checklist

- [x] This analysis module has a `README` and it is up to date.
- [ ] This analysis is recorded in the table in `analyses/README.md` and the entry is up to date.
- [x] The analytical code is documented and contains comments.

### CODE REVIEW

File: analyses/snv-frequencies/README.md [hunk 1]
```diff
@@ -115,9 +115,17 @@ curl -X GET "https://pedcbioportal.kidsfirstdrc.org/api/studies/ped_opentargets_
 
 To update the `case_set_id`s, rerun the `curl` command with your own PedcBioPortal web API access token. The token can be requested and downloaded at <https://pedcbioportal.kidsfirstdrc.org/webAPI>. More information about the access token is at <https://docs.cbioportal.org/2.2-authorization-and-authentication/authenticating-users-via-tokens#using-data-access-tokens>.
 
+Two more changes in column names and additions of three more columns are made for OT compatibility:
+ - `Gene_Ensembl_Id` is changed to `targetFromSourceId`
+ - `EFO` to `diseaseFromSourceMappedId`
+ Three additional columns added: 
+ - `chop_uuid` includes 32 digit UUIDs that are unique for each row of the file 
+ - `datatypeId` column with `somatic_mutation` in each row
+ - `datasourceId` column with `chop_gene_level_snv` in each row for gene-level table and `chop_variantlevel_snv` in each row for variant-level table.
```
Comment (zdorman) [inline analyses/snv-frequencies/README.md]: Add underscore to `chop_variantlevel_snv` in readme. Value is correct (`chop_variant_level_snv`) in the code.

**Review by zdorman — CHANGES_REQUESTED** [review]
Please correct `chop_variantlevel_snv` to `chop_variant_level_snv` in the readme, but all else looks good to me. Thanks!

**Comment by runjin326** [comment]
@zdorman, thanks so much for reviewing this! README.md has been updated accordingly.

**Review by zdorman — APPROVED** [review]
Thanks for the readme correction. All good!

**Pull request merged**